### PR TITLE
mw/kubernetes: rewrite README

### LINKED ISSUE #728: middleware/kubernetes: Improve/Expand README
Improve presentation of the kubernetes middleware readme, and expand content.

### PR DESCRIPTION
**Pull request by miekg**
Make the README comply with the other READMEs

Example our still left out.

Fixes #728 (some of it)

### CODE REVIEW

File: middleware/kubernetes/README.md [hunk 1]
```diff
@@ -26,163 +26,116 @@ Endpoint `A` records are constructed as "epname.myservice.mynamespace.svc.coredn
...
+  section containing the record for the question that actually had an answer.  This means that the
+  question and answer will not match. For example:
+
+   # host -v -t a google.com
+   Trying "google.com.default.svc.cluster.local"
+   ;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 50957
+   ;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0
+
+   ;; QUESTION SECTION:
+   ;google.com.default.svc.cluster.local. IN A
```
Comment (miekg) [inline middleware/kubernetes/README.md:116]: @chrisohaver @johnbelamaric I thought we were going to use CNAMEs for this?

**Comment by chrisohaver** [comment]
I already have a rewrite of the readme locally. Was going to do a PR today.

Sent from my mobile

On Jun 30, 2017, at 6:47 AM, Miek Gieben <notifications@github.com<mailto:notifications@github.com>> wrote:


@miekg commented on this pull request.

________________________________

In middleware/kubernetes/README.md<https://github.com/coredns/coredns/pull/763#discussion_r125012907>:

> +   ~~~
+   search ns1.svc.cluster.local svc.cluster.local cluster.local foo.com<http://foo.com>
+   ~~~
+
+  If no domains in the path produce an answer, a lookup on the bare question will be attempted.
+  A successful response will contain a question section with the original question, and an answer
+  section containing the record for the question that actually had an answer.  This means that the
+  question and answer will not match. For example:
+
+   # host -v -t a google.com<http://google.com>
+   Trying "google.com.default.svc.cluster.local"
+   ;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 50957
+   ;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0
+
+   ;; QUESTION SECTION:
+   ;google.com.default.svc.cluster.local. IN A


@chrisohaver<https://github.com/chrisohaver> @johnbelamaric<https://github.com/johnbelamaric> I thought we were going to use CNAMEs for this?

**Comment by chrisohaver** [comment]
>  I thought we were going to use CNAMEs for this?

It was part of the discussion, but we didn't land on it. I had suggested it as a work around of the mismatch name concern.

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/coredns] mw/kubernetes..." ]
>>  I thought we were going to use CNAMEs for this?
>

>It was part of the discussion, but we didn't land on it. I had suggested it as a work around of the mismatch name concern.

Ok. but the current impl. will not work. I've seen rewrite fail with Chrome that 
checks this - Chrome is likely not running in a k8s network, but this is just an 
example of a process that does more checking. The answer need to match the 
qname.

**Comment by chrisohaver** [comment]
If someone runs into this issue, they can disable autopath.  It's not enabled by default.
We could add the CNAME record answer later as optional behavior ... or as the default non-user configurable behavior if you are opposed to adding another configuration option.

Or would you prefer we implement the CNAME now (in 009)?

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/coredns] mw/kubernetes..." ]
>If someone runs into this issue, they can disable autopath.  It's not enabled by default.
>We could add the CNAME record answer later as optional behavior ... or as the default non-user configurable behavior if you are opposed to adding another configuration option.
>
>Or would you prefer we implement the CNAME now (in 009)?

Yes, we should; I'm afraid the current impl. is too fragile...

**Comment by chrisohaver** [comment]
Regarding the fact that I had already re-written the K8s readme... we need to coordinate better.
Perhaps we need a project mgmt tool to avoid wasted efforts.

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/coredns] mw/kubernetes..." ]
>Regarding the fact that I had already re-written the K8s readme... we need to coordinate better.
>Perhaps we need a project mgmt tool to avoid wasted efforts.

meh... I learned something and it wasn't a big deal. You assigned the issue to 
yourself... so I have only me to blame.

Please no mgmt tools

/Miek

--
Miek Gieben

**Pull request closed without merging**